mail-server/users.nix: don't expand variables in sieve script

eqyiel opened the pull request:
With this change the sieve script

```
require ["fileinto", "mailbox"];

if address :is "from" "notifications@github.com" {
  fileinto :create "GitHub";
  stop;
}

# This must be the last rule, it will check if list-id is set, and
# file the message into the Lists folder for further investigation
elsif header :matches "list-id" "<?*>" {
  fileinto :create "Lists";
  stop;
}

# Test: ''${1}
# Test: ${builtins.toString 1234};
```
...gets interpreted like this:

```
$ cat /var/sieve/ruben@maher.fyi/default.sieve                                                                                                                                                    
require ["fileinto", "mailbox"];

if address :is "from" "notifications@github.com" {
  fileinto :create "GitHub";
  stop;
}

# This must be the last rule, it will check if list-id is set, and
# file the message into the Lists folder for further investigation
elsif header :matches "list-id" "<?*>" {
  fileinto :create "Lists";
  stop;
}

# Test: ${1}
# Test: 1234;
```

Fixes #71

eqyiel committed:
mail-server/users.nix: don't expand variables in sieve script

eqyiel commented:
The CI error appears to be unrelated

r-raymond commented:
awesome, thanks a lot for fixing this! I'll merge if @The-M1k3y confirms it works for him

The-M1k3y commented:
Sorry, it took me some time to check. I can confirm that this fixes the issue on my server.
 (based on release 2.0.4)

eqyiel commented:
Rebased!

eqyiel commented:
Looks like Travis timed out while writing to the cache, apparently that can be configured though: https://docs.travis-ci.com/user/caching/#Setting-the-timeout

eqyiel committed:
ci: increase cache timeout to 10 minutes

The default is 3, and this can lead to cache timeouts while saving the contents
of /nix/store.

r-raymond commented:
It now fails with `The job exceeded the maximum time limit for jobs, and has been terminated.` But since this is unrelated to the PR, I'll merge.
